imports/ui/ranking/Ranking.js

```diff
@@ -1,21 +1,125 @@
+    /**
+     * Return the list of valid players, ranked by ELO and sorted by orderSort
+     * @param[in] format The Constants.GAME_TYPE game format
+     * @param[in] rankingSort The order to rank players by (currently always ELO)
+     * @param[in] orderSort The order to sort players by
+     * @returns A list of players with statistics
+     */
+    getOrderedPlayers(format, rankingSort, orderSort) {
+        return GameTypeUtils.getPlayers(format, rankingSort).map((p, i) => ({
             ...p,
             elo: p.elo.toFixed(3),
             rank: i + 1
-        }));
+        })).sort(orderSort.hash.orderSort);
+    },
+
+    /**
+     * List generator for ranking page statistics options
+     * @param[in] format The Constants.GAME_TYPE format
+     * @param[in] exclusions List of statistics to exclude by value (Ignore format!)
+     * @returns an object of formatted statistics keyed with statistics
+     */
+    getRankingStatistics(format, exclusions) {
+        let statisticsList = [
+            {value: "elo", format: "ELO"},
+            {value: "gamesPlayed", format: "Games"},
+            {value: "handWinRate", format: "Hand Win %"},
+            {value: "dealinRate", format: "Deal-in %"},
+            {value: "averageHandSize", format: "Avg Hand Size"},
+            {value: "averagePosition", format: "Avg Position"},
+            {value: "flyRate", format: "Bankrupt %"},
+            {value: "chomboTotal", format: "Chombos"}
+        ];
+
+        if (format === Constants.GAME_TYPE.JAPANESE) {
+            const japaneseStatisticsList = [
+                {value: "averageHandDora", format: "Avg Hand Dora"},
+                {value: "riichiRate", format: "Riichi %"},
+                {value: "riichiWinRate", format: "Riichi Win %"}
+            ];
+            statisticsList = statisticsList.concat(japaneseStatisticsList);
+        }
+
+        for (let stat of exclusions) {
+            let index = statisticsList.findIndex(i => i.value === stat.value);
+            if (index > -1) {
+                statisticsList.splice(index, 1);
+            }
+        }
+
+        return statisticsList;
+    },
+
+    /**
+     * Get statistic for a player, either sort or additional
+     * @returns the selected statistic for a player
+     */
+    getSortStatistic(player) {
+        return player[Session.get("sortStatistic")];
+    },
+    getAdditionalStatistic(player) {
+        return player[Session.get("additionalStatistic")];
+    },
+
+    /**
+     * Get formatted unit for a statistic if it has one
+     * @returns String of formatted unit or empty string
+     */
+    getSortStatUnit() {
+        return getStatisticUnit(Session.get("sortStatistic"));
+    },
+    getAdditionalStatUnit() {
+        return getStatisticUnit(Session.get("additionalStatistic"));
+    },
+
+    /**
+     * Exclusion lists for statistics options. Allows for manual entries
+     * @returns a list of statistics to exclude
+     */
+    sortRankingExclusion() {
```

Comment:
Are these the only places where `exclusions` can come from?

In that case I recommend having a list of possible statistics, and then picking items from it like so
```
const STATISTICS_LIST = [...];
additionalRankingExclusion() {
  return STATISTICS_LIST[2];
}
```

or something like this, it'll make the comparisons easier to read since you'll be able to just compare references to objects instead of the inner content of these objects. It's also more performant but that's whatever.

It's kind of a big change and this is fine though

Reply:
I'm going to leave this for now.